packages/core/src/services/evaluationResults/fetchEvaluationResultsByDocumentLogs.ts

```diff
@@ -0,0 +1,45 @@
+import { database } from '../../client'
+import { Result } from '../../lib'
+import {
+  EvaluationResultsRepository,
+  EvaluationsRepository,
+  ResultWithEvaluation,
+} from '../../repositories'
+
+export async function fetchEvaluationResultsByDocumentLogs(
+  {
+    workspaceId,
+    documentLogIds,
+  }: {
+    workspaceId: number
+    documentLogIds: number[]
+  },
+  db = database,
+) {
+  documentLogIds = [...new Set(documentLogIds)]
+
+  const evaluationsRepository = new EvaluationsRepository(workspaceId, db)
+  const resultsRepository = new EvaluationResultsRepository(workspaceId, db)
+
+  const results = await resultsRepository
+    .findByDocumentLogIds(documentLogIds)
+    .then((r) => r.unwrap())
+
+  const evaluations = await evaluationsRepository
+    .filterById([...new Set(results.map((r) => r.evaluationId))])
+    .then((r) => r.unwrap())
+
+  const resultsByDocumentLog = documentLogIds.reduce<
+    Record<number, ResultWithEvaluation[]>
+  >((acc, documentLogId) => {
+    acc[documentLogId] = results
+      .filter((r) => r.documentLogId === documentLogId)
+      .map((result) => ({
+        result: result,
+        evaluation: evaluations.find((e) => e.id === result.evaluationId)!,
+      }))
+    return acc
+  }, {})
```

Comment:
Not sure this is the best way to do this. Maybe it is. Just wondering if could be done in another way.

Reply:
The data structure it returns is an object where the keys are documentLogIds and the values are Lists of EvaluationResults. If I were to do this as a list it would be like a matrix, and that is unnecessarily complex to work with/find things. (I think)